File: Green.Core.Tests/InterpreterTests.cs
using System;
using Xunit;

namespace Green.Tests
{
    public sealed class InterpreterTests
    {
        private readonly Interpreter _interpreter = new Interpreter();

        [Fact]
        public void Eval_Number()
        {
            var result = _interpreter.EvalSource("10");
            Assert.Equal(10L, result);
        }

        [Fact]
        public void Eval_Add()
        {
            var result = _interpreter.EvalSource("(+ 2 3)");
            Assert.Equal(5L, result);
        }

        [Fact]
        public void Eval_NestedAdd()
        {
            var result = _interpreter.EvalSource("(+ (+) (+ 2 3) (+ 4 5 6))");
            Assert.Equal(20L, result);
        }
    }
}

File: Green.Core.Tests/InterpreterTests.fs
module InterpreterTests

open Xunit
open Green.Interpreter
open Green.Obj

[<Fact>]
let Eval_Number() =
    let interpreter = Interpreter()
    let result = interpreter.EvalSource("10")
    Assert.Equal(Value.ofInt 10L, result)

[<Fact>]
let Eval_Add() =
    let interpreter = Interpreter()
    let result = interpreter.EvalSource("(+ 2 3)")
    Assert.Equal(Value.ofInt 5L, result)

[<Fact>]
let Eval_NestedAdd() =
    let interpreter = Interpreter()
    let result = interpreter.EvalSource("(+ (+) (+ 2 3) (+ 4 5 6))")
    Assert.Equal(Value.ofInt 20L, result)
